design/sketch.webidl

```diff
@@ -573,15 +573,24 @@ dictionary WebGPUCommandBufferDescriptor {
 // ****************************************************************************
 
 // Fence
+
+dictionary WebGPUFenceDescriptor {
+    WebGPUQueue signalQueue = null;
+    u64 initialValue = 0;
+};
+
 interface WebGPUFence {
-    bool wait(Number milliseconds);
-    readonly attribute Promise<void> promise;
+    u64 getCompletedValue();
+    Promise<void> onCompletion(u64 completionValue);
 };
 
 // Queue
 interface WebGPUQueue {
     void submit(sequence<WebGPUCommandBuffer> buffers);
-    WebGPUFence insertFence();
```

Comment:
The whole buffer-mapping.md doc is broken now because we don't have transition, we need to rework it completely, and figure out how it will work with the updated fences, if at all. Since buffer mapping isn't an enqueued operation anymore, I don't think it makes sense to wait on a fence to see the mapping.

Reply:
Since this PR makes it out-of-date, I was expecting you to update the buffer mapping document as part of this PR.

We can do it as a subsequent update if you wish but we should tackle it sooner rather than later.

Reply:
At the [F2F](https://docs.google.com/document/d/1NqJWi_VW7njO8AeMmJbqh5HyXi1mkNY7XlJFPjtWYtg) we discussed buffer mapping again and seemed to have consensus around:
 - We need `WebGPUBuffer.mapReadAsync` and `WebGPUBuffer.mapWriteAsync`.
 - `mapWrite*` returns a zero-filled buffer.
 - The buffer is either owned by the GPU (on one or multiple queues) or owned by the CPU. Never both at the same time.

Things that were still unknowns:
 - How do we know when the mapping is ready? We talked about promises but fell in the same debate of promises vs. not as a year ago.
 - Are we able to map sub-ranges of the buffer, or only the whole range?

At this point I think our [old proposal](https://docs.google.com/document/d/1HFzMMvDGHFtTgjNT0j-0SQ1fNU9R7woZ4JuNJdAXBjg) of having a `WebGPUMappedMemory` object might make sense (minus the transitions).

I agree that buffer mapping needs to be handled sooner than later, but I think it is way bigger than the scope of this PR.

Reply:
Would it make sense to have a special pass or queue item that does buffer transfer? That way it can be submitted and ordered amongst the work.

Reply:
Done